glue/core/fixed_resolution_buffer.py

```diff
@@ -143,24 +143,41 @@ def compute_fixed_resolution_buffer(data, bounds, target_data=None, target_cid=N
     if cache_id is not None:
 
         if subset_state is None:
-            # Use uuid for component ID since otherwise component IDs don't return
+            # Use uuid for data and component ID since otherwise component IDs don't return
             # False when comparing two different CIDs (instead they return a subset state).
             # For bounds we use a special wrapper that can identify wildcards.
-            current_array_hash = (data, bounds, target_data, target_cid.uuid, broadcast)
+            current_array_hash = (data.uuid, bounds, target_data.uuid, target_cid.uuid, broadcast)
         else:
-            current_array_hash = (data, bounds, target_data, subset_state, broadcast)
+            current_array_hash = (data.uuid, bounds, target_data.uuid, subset_state, broadcast)
 
-        current_pixel_hash = (data, target_data)
+        current_pixel_hash = (data.uuid, target_data.uuid)
 
         if cache_id in ARRAY_CACHE:
             if ARRAY_CACHE[cache_id]['hash'] == current_array_hash:
                 return ARRAY_CACHE[cache_id]['array']
 
-        # To save time later, if the pixel cache doesn't match at the level of the
-        # data and target_data, we just reset the cache.
-        if cache_id in PIXEL_CACHE:
-            if PIXEL_CACHE[cache_id]['hash'] != current_pixel_hash:
-                PIXEL_CACHE.pop(cache_id)
+        # For pixel transformations, we don't necessarily need the cache_id to match, as coordinate
+        # transformations can be the same for different callers - for instance, layer artists for
+        # subsets might need the same transformations over all subsets and the same as for the actual
+        # data. Therefore, instead of extracting the cache_id entry from PIXEL_CACHE, we instead loop
+        # over PIXEL_CACHE, and for each pixel component ID, we find the first entry that matches the
+        # hash and the bounds, since that should uniquely identify the transformation. Note that we
+        # still use cache_id because we still want to make sure we save results from different layers
+        # to the cache in case any layers get removed or changed.
+        matching_pixel_cache = {}
+        for chid, i_cache in PIXEL_CACHE.items():
```

Comment:
actually I am surprised Ruff does not complain now that `chid` is never used – or suggested to use `PIXEL_CACHE.values()` in the first place... 😂

Reply:
Oh yeah, we should fix that, we should just use ``.values()``...